File: app/src/main/java/com/semisonfire/cloudgallery/ui/main/trash/TrashContract.java
package com.semisonfire.cloudgallery.ui.main.trash;

import com.semisonfire.cloudgallery.data.model.Photo;
import com.semisonfire.cloudgallery.ui.base.MvpPresenter;
import com.semisonfire.cloudgallery.ui.base.MvpView;

import java.util.List;

public interface TrashContract {

    interface View extends MvpView {

        void onTrashLoaded(List<Photo> photos);

        void onPhotoDeleted(Photo photo);

        void onPhotoRestored(Photo photo);

        void onTrashCleared();

    }

    interface Presenter<V extends View> extends MvpPresenter<V> {

        void getPhotos(int page);

        void restorePhotos(List<Photo> photos);

        void deletePhotos(List<Photo> photos);

        void clear();
    }

}

File: app/src/main/java/com/semisonfire/cloudgallery/ui/main/trash/TrashContract.kt
package com.semisonfire.cloudgallery.ui.main.trash

import com.semisonfire.cloudgallery.data.model.Photo
import com.semisonfire.cloudgallery.core.mvp.MvpPresenter
import com.semisonfire.cloudgallery.core.mvp.MvpView

interface TrashContract {

  interface View : MvpView {

    fun onTrashLoaded(photos: List<Photo>)

    fun onPhotoDeleted(photo: Photo)

    fun onPhotoRestored(photo: Photo)

    fun onTrashCleared()

  }

  interface Presenter : MvpPresenter<View> {

    fun getPhotos(page: Int)

    fun restorePhotos(photos: List<Photo>)

    fun deletePhotos(photos: List<Photo>)

    fun clear()
  }

}
